lib/autoloads/youtube_url_formatter.rb

```diff
@@ -8,4 +8,12 @@ def self.format(url)
     youtube_id_match = YOUTUBE_ID_REGEX.match(url)
     "https://www.youtube.com/embed/#{youtube_id_match[1]}" if youtube_id_match
   end
+
+# youtube動画URLから動画IDのみ返す関数(中身はself.formatと同じ)
+  def self.format_ID(url)
+    src_match = SRC_REGEX.match(url)
+    url = src_match[1] if src_match
+    youtube_id_match = YOUTUBE_ID_REGEX.match(url)
+    "#{youtube_id_match[1]}" if youtube_id_match
```

Comment:
こちらでよいと思います。あと，現状ですと2回目に保存した際にchannel IDが nil にならないでしょうか。

```suggestion
    youtube_id_match[1] if youtube_id_match
```